lib/phoenix_live_view/channel.ex

```diff
@@ -62,6 +82,24 @@ defmodule Phoenix.LiveView.Channel do
     e -> reraise(e, __STACKTRACE__)
   end
 
+  def handle_info({:EXIT, pid, reason} = msg, state) do
+    case Map.fetch(all_asyncs(state), pid) do
```

Comment:
I am worried `all_asyncs` may be too expensive. I will have to think of a better way to do this. :( Ping me later, we can try once again to use the monitor trick I mentioned.

Reply:
Here is the idea:

```elixir
cid = 123

{:ok, pid} =
  Task.start_link(fn ->
    receive do
      {:context, ref} ->
        try do
          # do stuff and then send to the ref,
          # that will automatically disable the monitor
          send(ref, {:ok, Process.sleep(:infinity)})
        catch
          kind, reason ->
            # use to_exit
            send(ref, {:exit, reason})
            :erlang.raise(kind, reason, __STACKTRACE__)
        end
    end
  end)

ref = :erlang.monitor(:process, pid, alias: :reply_demonitor, tag: {Phoenix.LiveView.Async, cid})
send(pid, {:context, ref})

# Emulate a broken link with trapped exits
Process.flag(:trap_exit, true)
Process.exit(pid, :kill)

receive do
  {{Phoenix.LiveView.Async, cid}, ^ref, _, _, reason} -> IO.inspect({cid, ref, reason})
end
```

Instead of relying on exits, we use the monitor message. We use `alias: :reply_demonitor` to kill the monitor as soon as you send a message to `ref` (which you want on both try/catch branches). And we use the `:tag` option so we pass the cid with the "down" message, so you no longer need `all_asyncs`. You can include more information in the tag if necessary.